database/database.go

```diff
@@ -0,0 +1,183 @@
+	compID, err := driver.FindComponentID(&componentStatus.ID)
+	if err != nil {
+		return fmt.Errorf("error getting component ID: %w", err)
+	}
+
+	percentiles, err := json.Marshal(componentStatus.LatencyPercentiles)
+	if err != nil {
+		return fmt.Errorf("error marshaling latency percentiles: %w", err)
+	}
+
+	insertQuery := `INSERT INTO v9.public.stats
+    (worker_id, component_id, color, stat_window_seconds, hits,
+     avg_response_bytes, avg_ms_latency, ms_latency_percentiles)
+    VALUES ($1, $2, $3, $4, $5, $6, $7, $8)`
+
+	_, err = driver.db.Exec(
+		insertQuery, workerID, compID, componentStatus.Color, componentStatus.StatWindow,
+		componentStatus.Hits, componentStatus.AvgResponseBytes, componentStatus.AvgMsLatency, string(percentiles))
+	if err != nil {
+		return fmt.Errorf("error sending stats to database: %w", err)
+	}
+
+	return nil
+}
+
+// TODO: Refactor to make cleaner
+func (driver *Driver) InsertLog(workerID string, compLog worker.ComponentLog) error {
+	compDBID, err := driver.FindComponentID(&compLog.ID)
+	if err != nil {
+		return fmt.Errorf("error getting comp id for logs: %w", err)
+	}
+
+	// Get data to write on top of
+	var logID *string
+	var logText *string
+	var logError *string
+
+	getOriginalQuery := `SELECT log_id, log_text, log_error FROM v9.public.logs
+	WHERE worker_id = $1 AND component_id = $2 AND execution_num = $3`
+
+	err = driver.db.QueryRow(getOriginalQuery, workerID, compDBID, compLog.DedupNumber).Scan(&logID, &logText, &logError)
+	if err != nil && err != sql.ErrNoRows {
+		return fmt.Errorf("error getting previous logs from database: %w", err)
+	}
+
+	if logID == nil {
+		randomUUID, uuidErr := uuid.NewRandom()
+		if uuidErr != nil {
+			return fmt.Errorf("error creating a random UUID for the log table: %w", uuidErr)
+		}
+		randomID := randomUUID.String()
+		logID = &randomID
+	}
+
+	if compLog.Log != nil {
+		logText = compLog.Log
+	}
+
+	if compLog.Error != nil {
+		logError = compLog.Error
+	}
+
+	insertUpdateQuery := `INSERT INTO v9.public.logs
+    (log_id, worker_id, component_id, execution_num, log_text, log_error, received_time)
+    VALUES ($1, $2, $3, $4, $5, $6, NOW())
+    ON CONFLICT (log_id) DO UPDATE SET log_text = $5, log_error = $6, received_time = NOW()`
+
+	_, err = driver.db.Exec(insertUpdateQuery, logID, workerID, compDBID, compLog.DedupNumber, logText, logError)
+	if err != nil {
+		return fmt.Errorf("error doing final log database update: %w", err)
+	}
+	return err
+
+}
+
+func (driver *Driver) EnterDeploymentEntry(compID *worker.ComponentID) error {
+	compDBID, err := driver.FindComponentID(compID)
+	if err != nil {
+		return err
+	}
+
+	var deploymentStartTime string
+	upsertQuery := `INSERT INTO v9.public.deploying(component_id, deployment_reason)
+	VALUES ($1, $2) ON CONFLICT DO NOTHING
+	RETURNING to_char(deployment_start_time::timestamp at time zone 'UTC', 'YYYY-MM-DD"T"HH24:MI:SS"Z"')`
```

Comment:
How does this correspond to TIMESTAMPTZ? It looks like you're using UTC, will this cause issues?

Reply:
That's just for error reporting, the format doesn't really matter